### PR TITLE
[LOOP-1365] Refactor basal schedule syncing in DashPumpManager

### PR DESCRIPTION
**Pull request by mpangburn**
Refactoring to support https://tidepool.atlassian.net/browse/LOOP-1365 for DashPumpManager

### CODE REVIEW

File: DashKitUI/PumpManager/DashPumpManager+UI.swift [hunk 1]
```diff
@@ -66,11 +66,12 @@ extension DashPumpManager {
 extension DashPumpManager {
 
     public func syncScheduleValues(for viewController: BasalScheduleTableViewController, completion: @escaping (SyncBasalScheduleResult<Double>) -> Void) {
-        setBasalSchedule(dailyItems: viewController.scheduleItems) { (error) in
-            if let error = error {
+        syncBasalRateSchedule(items: viewController.scheduleItems) { result in
+            switch result {
+            case .success(let schedule):
+                completion(.success(scheduleItems: schedule.items, timeZone: schedule.timeZone))
+            case .failure(let error):
```
Comment (darinkrauss) [inline DashKitUI/PumpManager/DashPumpManager+UI.swift:73]: [nit] Do we have any team coding standards about order of `.failure` and `.success` in `switch`? I know it doesn't matter at all, but I've always done failure first (just for consistency). It isn't a big deal so don't bother to change, just curious. 

cc: @ps2

**Review by darinkrauss — APPROVED** [review]
LGTM! 🚜

**Review by ps2 — APPROVED** [review]
LGTM!

File: DashKitUI/PumpManager/DashPumpManager+UI.swift (hunk 1 above)
Comment (mpangburn) [reply DashKitUI/PumpManager/DashPumpManager+UI.swift:73]: I default to `success` first but don't have a strong opinion.

**Pull request merged**